src/shared/cline/banner.ts

```diff
@@ -0,0 +1,238 @@
+/**
+ * Banner data structure for backend-to-frontend communication.
+ * Backend constructs this JSON, frontend renders it via BannerCarousel.
+ */
+
+export interface BannerCardData {
+	/** Unique identifier for the banner (used for dismissal tracking) */
+	id: string
+
+	/** Banner title text */
+	title: string
+
+	/** Banner description/body text */
+	description: string
+
+	/**
+	 * Icon ID from Lucide icon set (e.g., "lightbulb", "megaphone", "terminal")
+	 * LINK: https://lucide.dev/icons/
+	 * Optional - if omitted, no icon is shown
+	 */
+	icon?: string
+
+	/**
+	 * Severity level determines styling
+	 * - "info" (default): informational
+	 * - "danger": warning/error state
+	 * - "secondary": less prominent
+	 */
+	severity?: "info" | "danger" | "secondary"
+
+	/**
+	 * Optional inline action displayed at the end of the description text
+	 * Rendered as a link or button in the text flow
+	 */
+	endAction?: BannerAction
+
+	/**
+	 * Optional footer action buttons
+	 * Rendered below the description as prominent buttons
+	 */
+	actions?: BannerAction[]
+
+	/**
+	 * If true, only show banner to users logged in with Cline account
+	 * If false, only show to non-Cline users
+	 * If undefined, show to all users
+	 */
+	clineUserOnly?: boolean
+
+	/**
+	 * Platform filter - only show on specified platforms
+	 * If undefined, show on all platforms
+	 */
+	platforms?: ("windows" | "mac" | "linux")[]
```

Comment:
Could we use `macos` here to be consistent with the current backend implementation.